pallets/pools/src/tranche.rs

```diff
@@ -225,6 +260,41 @@ where
 		self.accrue(now)?;
 		Ok(self.debt)
 	}
+
+	pub fn create_asset_metadata(
+		&self,
+		decimals: u32,
+		parachain_id: ParachainId,
+		token_name: Vec<u8>,
+		token_symbol: Vec<u8>,
+	) -> AssetMetadata<Balance, CustomMetadata>
+	where
+		Balance: Zero,
+		Currency: Encode,
+		CustomMetadata: Parameter + Member + TypeInfo,
+	{
+		let tranche_id =
+			WeakBoundedVec::<u8, ConstU32<32>>::force_from(self.currency.encode(), None);
```

Comment:
Not sure if we should go with it this way. We will never notice if this fails. At least we need a test to cover a failure of this, where we do a `from` instead of `force_from`.

also: 
* why is it bounded by length of `32`?

We could enforce the `CurrencyId` to implement some trait we define. Like `trait GenerelKey`. And then we define this manually for the `enum CurrencyId` we provide. 

@NunoAlexandre could this be useful for xcm in general?

Reply:
I need to go through this in more detail as I am just scanning it right now, but for xcm purposes all we need is to have a way of identifying a tranche token uniquely; if it's with `GeneralKey` or `GeneralIndex` or another `Junction`, it doesn't matter much; it just has to be unique and preferrably deterministically derived from the pool and/or tranche_id.

Reply:
We use `WeakBoundedVec<>::force_from` also in `runtime/common/impls.rs`, this is why I thought it is ok to use it here as well. Is this still a concern or can we merge this PR? @mustermeiszer @NunoAlexandre

Reply:
At least from what I see, we use this force for only [u8;2] arrays. But the encoding is arbitrarily large and we will not notice the changes.

Reply:
But I will bring in a proposal for letting the currency define its general key itself, then we can use this method. At least for now, I would not block this anymore.